web-client/public/js/setup-load-and-import-handlers.js

```diff
@@ -171,8 +171,8 @@ export const setupLoadAndImportHandlers = (grnState) => {
      * for helping to resolve this.
      */
 
-    // $(".upload").change(uploadHandler(loadGrn));
     $("body").on("change", ".upload", uploadHandler(loadGrn));
```

Comment:
Looks like this just goes back to what it was, but quick question now that I see the commented-out line: are you aware of any difference between the top and bottom approaches? It seems like they would be the same, but clearly someone historically thought that doing `$("body").on("change", .....` would be better. I guess we can look up the commit log

Reply:
The two approaches are similar, but the `.change` method is deprecated in jQuery from version 3.3, so Ona reconstructed the code to address this. The first approach attempts to bind the event directly to existing elements with the class upload, meaning these elements should already be present in the HTML file. In contrast, the second approach is better because it uses event delegation by binding the `change` event to the body element and specifying `.upload` as the target. This allows for handling events for upload elements that currently exist or will be added to the DOM in the future, meaning we can dynamically add the `upload` class via JavaScript and still manage the events effectively.

Reply:
Ah got it—well deprecation is certainly a good reason for changing things, plus yes, the newer version indeed does “late binding” on the selector 👌🏽 Thanks for tracking this down!